From 2954ab8983fe9122bf4ef4263d6b547b99b806da Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Sat, 7 Oct 2017 18:59:06 +0200
Subject: [PATCH 15/15] Destroy kref_debug on failure in drbd_create_device.

If we fail with ERR_NOMEM, drbd_create_device will be called
again and crash if we do not clean up kref_debug.
---
 drbd/drbd_main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drbd/drbd_main.c b/drbd/drbd_main.c
index d040b88..443ccbb 100644
--- a/drbd/drbd_main.c
+++ b/drbd/drbd_main.c
@@ -3688,6 +3688,7 @@ out_no_disk:
 	blk_cleanup_queue(q);
 out_no_q:
 	kref_put(&resource->kref, drbd_destroy_resource);
+	kref_debug_destroy(&device->kref_debug);
 	kfree(device);
 out_no_device:
 	bdput(block_device);
-- 
2.8.0-rc4

